### PR TITLE
[feat]: Adding a new sparsifying transform using sklearn

### PR DESCRIPTION
**Pull request by hanaecarrie**
Adding plugg-in for Dictionary Learning based on the sklearn library only.

This is a 2-step based apporach to reconstruct MRI data.
We use a dictionary (sparsifying tranform) learnt offline on a MRI database in the reconstruction instead of using Wavelets.

You must provide an opensource training database. 

The code in the Conda-Vu reconstruction is slow. (~1 hour for 10 iterations).

### CODE REVIEW

**Comment by hanaecarrie** [comment]
cannot rerun the tests on the neurospin/pisap repo without a new PR.
Need help :/ @AGrigis

**Comment by sfarrens** [comment]
Try using `travis_retry pip install astropy`

File: examples/mri/reconstruction_with_DL.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+#############################################################################
+# Load the database to train the dictionary
+# -----------------------------------------
+#
+# The database should be a list of list of 2D MRI images grouped by subjects.
+# In this tutorial, the database contains only brain images.
+# Besides, all images have same size and all subjects have the same
+# number of slices.
+
+database_path = '/home/hc253658/STAGE/data_DL/training_database.npy'
```
Comment (sfarrens) [inline examples/mri/reconstruction_with_DL.py]: Does anyone have access to this file?

**Comment by hanaecarrie** [comment]
The issue is that this file contains a training database from Lisa Leroi (CEA) and I am absolutely not sure that we can get open access to her data…
I was told to let this like that and that you will figure out what is possible to do.

If we cannot think of putting the data in the package, we can put only the trained model and make the reconstruction with it and add some comments about how the model was  trained.

Otherwise, I put all my data in the folder /neurospin/optimed/hanae…
There is a README in my folder to explain the structure.
If you don’t manage to find it, let me know, I’ll help you out.

Hope this can help.
Hanaé

**Comment by AGrigis** [comment]
Before merging, we need to update the code in order to work with modopt rather than sf_tools.

**Pull request merged**